vignettes/how-to-update-released-site.Rmd

```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
-* Create and checkout a branch based on the tag of your most recent GitHub/CRAN
-  release.
-  
-  Example for readr, whose latest release is 2.1.1:  
-  `git checkout -b update-pkgdown-2-1-1 v2.1.1`
-  
-  General pattern:  
-  `git checkout -b NEW-BRANCH-NAME NAME-OF-RELEASE-TAG`
-* Backport innovations from the future. Example of a useful opening move:  
-  `git checkout main -- _pkgdown.yml .github/workflows/pkgdown.yaml`
-
-  Most likely candidates:
-  - `_pkgdown.yml`
-  - `.github/workflows/pkgdown.yaml`
-  - `Config/Needs/website: tidyverse/tidytemplate` in DESCRIPTION
-  - Any fixes to `README.Rmd` / `README.md` or released roxygen comments. Don't
-    forget to `devtools::build_readme()` or `devtools::document()`, in that
-    case.
-  
-  These are the two main Git commands you'll need:
-  ```
-  git checkout REF -- path/to/a/specific/file
-  git cherry-pick SHA
-  ```
-* Monitor your progress locally with `pkgdown::build_site()`.
-* Push this branch, maybe using `usethis::pr_push()`. Don't bother opening
-  a pull request (the branch will still be created and pushed).
-* Go to your package's GHA page, maybe using `usethis::browse_github_actions()`.
-  Select the pkgdown workflow. Click *Run workflow* and select the branch you
-  just pushed. If there's no dropdown menu for this, that means your pkgdown
-  workflow config is not current.
-* Scrutinize your newly published **released** site and make sure things look
-  right.
-* You can let this branch sit around for a while, in case you need to iterate
-  on this or if you'd like to backport more niceties before your next CRAN
-  release. Or you can delete immediately, if you're happy with the released
-  site.
+This vignette shows you how to update the released version of your site to match the dev version of the site, so the first step is to ensure that the dev site looks the way that you want it.
+
+## Process
+
+If you're up to speed with the basic idea and just want some code to follow here it us.
+Otherwise, read more below.
+
+### Setup
+
+First, make sure you're in the main branch, and you have the latest version:
+
+```{r}
+gert::git_branch_checkout("main")
+gert::git_pull()
+```
+
+Next figure out the released version that we're updating:
+
+```{r}
+ver <- desc::desc_get_version()[1, 1:3]
+```
+
+We'll use this to create the branch that you'll work in:
+
+```{r}
+gert::git_branch_create(paste0("pkgdown-v", ver), paste0("v", ver))
+```
+
+### Backport changes
+
+Now you need to backport changes from the dev site into this branch.
+Run this R code to generate the git code to pull changes for the most common locations:
+
+```{r}
+files <- c(
+  # overall site config 
+  "_pkgdown.yml",
+  # the GHA that'll impenet
+  ".github/workflows/pkgdown.yaml",
+  # readme and vignettes
+  "README.md", "vignettes",
+  # logo and favicon
+  "man/figures/", "pkgdown/", 
+  # Author metadata and Config/Needs/Website
+  "DESCRIPTION"
+)
+
+glue::glue("git checkout v{ver} -- {files}")
```

Comment:
Feels like you still need to explicitly say to *run* that git command in the terminal.

I assume you checked/experienced this, but you can just put a whole folder there in `git checkout`? I think I always do this move with individual files 🤔

Reply:
I think the reader of this doc can be assumed to understand that if I'm generating git code in R, they'll need to copy and paste it.

And yes, you can indeed checkout folders.